app-service-template/functions/sample.go

```diff
@@ -96,6 +97,61 @@ func (s *Sample) LogEventDetails(ctx interfaces.AppFunctionContext, data interfa
 	return true, event
 }
 
+// SendCommand is example of how to use the CommandClient to send commands to devices
+func (s *Sample) SendCommand(ctx interfaces.AppFunctionContext, data interface{}) (bool, interface{}) {
+	lc := ctx.LoggingClient()
+	lc.Debugf("SendCommand function called in pipeline '%s'", ctx.PipelineId())
+
+	if data == nil {
+		// Go here for details on Error Handle: https://docs.edgexfoundry.org/latest/microservices/application/ErrorHandling/
+		return false, fmt.Errorf("function SendCommand in pipeline '%s': No Data Received", ctx.PipelineId())
+	}
+
+	event, ok := data.(dtos.Event)
+	if !ok {
+		return false, fmt.Errorf("function SendCommand in pipeline '%s', type received is not an Event", ctx.PipelineId())
+	}
+
+	lc.Debugf("Issuing Command Query for device %s in pipeline '%s'", event.DeviceName, ctx.PipelineId())
+
+	// First get the list of commands available for the Device that this current Event is for
+	response, err := ctx.CommandClient().DeviceCoreCommandsByDeviceName(context.Background(), event.DeviceName)
+	if err != nil {
+		return false, fmt.Errorf("failed to get list of commands for %s device: %v in pipeline '%s'", event.DeviceName, err, ctx.PipelineId())
+	}
+
+	var commandName string
+
+	lc.Debugf("Device %s has %d commands to choose from. (in pipeline '%s')", event.DeviceName, len(response.DeviceCoreCommand.CoreCommands), ctx.PipelineId())
+
+	for _, command := range response.DeviceCoreCommand.CoreCommands {
+		if command.Get {
+			commandName = command.Name
+			break
```

Comment:
does this mean we issue the first Get command in the list?

Reply:
Yes, random Get command to demonstrate how to Query and the Send the Get command.

Reply:
Add comments